lib/Dialect/TTNN/Transforms/TTNNWorkarounds.cpp

```diff
@@ -479,13 +554,16 @@ class TTNNWorkarounds : public impl::TTNNWorkaroundsBase<TTNNWorkarounds> {
   void runOnOperation() final {
     if (decompositionWorkaroundsEnabled) {
       // Placeholder for workaround decomposition patterns.
+
+      GreedyRewriteConfig config = GreedyRewriteConfig();
+      config.useTopDownTraversal = true;
+      config.maxIterations = 1;
```

Comment:
I had the same issue for the reduce workarounds, so I've extracted the configuration setup into separate function, which I call for both layout and decomposition workarounds, please take a look [here](https://github.com/tenstorrent/tt-mlir/pull/1625/files#diff-5157bf289d6d303570d1486babf360c6dfcf9a2af2e67ce3b157992ac5b28267). Currently we don't need a different setup for those workarounds, I think two iterations will work for you too, if we need to do so in the future we can pass additional arguments to that function.

Reply:
Thanks for looking into this, yeah I just made a quick fix for this draft PR, but I think long term we should allow each workaround to setup the config as needed. I will make that change as this PR will probably merge after yours.